src/core/ext/transport/chttp2/client/insecure/channel_create.c

```diff
@@ -64,14 +66,37 @@ static grpc_channel *client_channel_factory_create_channel(
     const char *target, grpc_client_channel_type type,
     const grpc_channel_args *args) {
   // Add channel arg containing the server URI.
+  // Firstly, let's make sure \a target is a valid URI.
+  grpc_uri *target_uri = NULL;
+  char *target_uri_str = NULL;
+  grpc_uri_section uri_section = GRPC_URI_SECTION_NONE;
+  if ((target_uri = grpc_uri_parse(target, true /* suppress errors */,
+                                   &uri_section)) == NULL) {
+    if (uri_section == GRPC_URI_SECTION_SCHEME) {
+      // If the parsing error was due to the lack of scheme, try to fix that by
+      // prepending the default resolver prefix.
+      gpr_asprintf(&target_uri_str, "%s%s",
+                   grpc_resolver_registry_get_default_prefix(), target);
+      // try again...
+      if ((target_uri = grpc_uri_parse(
+               target_uri_str, true /* suppress errors */, NULL)) == NULL) {
+        gpr_free(target_uri_str);
+        return NULL;
+      }
+    } else {
+      return NULL;
+    }
+  }
+  grpc_uri_destroy(target_uri);
```

Comment:
I don't like the code duplication (secure/insecure versions). I'll likely place this code in some util file (suggestions welcome). I wanted to put the review out anyway though.